### PR TITLE
Accept parameters by reference/const reference in few functions.

### PR DESCRIPTION
**Pull request by omtcyfz**
This PR changes few functions to accept parameters by reference/const reference while preventing unnecessary copying.

### CODE REVIEW

File: modules/imgproc/include/opencv2/imgproc.hpp [hunk 1]
```diff
@@ -1113,13 +1113,13 @@ class CV_EXPORTS_W Subdiv2D
 protected:
     int newEdge();
     void deleteEdge(int edge);
-    int newPoint(Point2f pt, bool isvirtual, int firstEdge = 0);
+    int newPoint(Point2f &pt, bool isvirtual, int firstEdge = 0);
```
Comment (sergiud) [inline modules/imgproc/include/opencv2/imgproc.hpp]: Why not const reference?

File: modules/imgproc/include/opencv2/imgproc.hpp [hunk 2]
```diff
@@ -1113,13 +1113,13 @@ class CV_EXPORTS_W Subdiv2D
...
     void deletePoint(int vtx);
     void setEdgePoints( int edge, int orgPt, int dstPt );
     void splice( int edgeA, int edgeB );
     int connectEdges( int edgeA, int edgeB );
     void swapEdges( int edge );
-    int isRightOf(Point2f pt, int edge) const;
+    int isRightOf(Point2f &pt, int edge) const;
```
Comment (sergiud) [inline modules/imgproc/include/opencv2/imgproc.hpp]: Here as well, why not const reference?

File: modules/imgproc/src/subdivision2d.cpp [hunk 3]
```diff
@@ -215,7 +215,7 @@ static double triangleArea( Point2f a, Point2f b, Point2f c )
     return ((double)b.x - a.x) * ((double)c.y - a.y) - ((double)b.y - a.y) * ((double)c.x - a.x);
 }
 
-int Subdiv2D::isRightOf(Point2f pt, int edge) const
+int Subdiv2D::isRightOf(Point2f &pt, int edge) const
```
Comment (sergiud) [inline modules/imgproc/src/subdivision2d.cpp]: Here too, why not const reference?

File: modules/imgproc/src/subdivision2d.cpp [hunk 4]
```diff
@@ -251,7 +251,7 @@ void Subdiv2D::deleteEdge(int edge)
     freeQEdge = edge;
 }
 
-int Subdiv2D::newPoint(Point2f pt, bool isvirtual, int firstEdge)
+int Subdiv2D::newPoint(Point2f &pt, bool isvirtual, int firstEdge)
```
Comment (sergiud) [inline modules/imgproc/src/subdivision2d.cpp]: Why not const reference?

**Comment by omtcyfz** [comment]
@sergiud You're totally right. Apologies for not doing it in the first place.

**Comment by sergiud** [comment]
@omtcyfz Looks good now. I would just suggest squashing all three commits into a single one.

**Comment by vpisarev** [comment]
signature of Subdiv2D methods, as well as any public API in the main opencv repository may not be changed. The nearest point when it can be changed is OpenCV 4.0, which is 1-2 years away from now. Also, for compact structures like Point2f passing reference instead of the structure itself does not have much sense. You won't save any processing time

**Comment by vpisarev** [comment]
let's close it for now, because it break binary compatibility and the proposed fixes are not essential.

**Pull request closed without merging**